query-connector/src/app/layout.tsx

```diff
@@ -23,33 +22,31 @@ export default async function RootLayout({
 }) {
   // Initializes user session on server side for the first load
   // if session does not exists then session object remains null
-  await auth();
+  const session = await auth();
 
   const runtimeConfig = {
     AUTH_DISABLED: process.env.AUTH_DISABLED || "false",
   };
 
   return (
     <html lang="en">
-      <SessionProvider>
-        <body>
-          <ToastContainer
-            position="bottom-left"
-            icon={false}
-            stacked
-            hideProgressBar
-          />
-          <div className="application-container">
+      <body>
+        <ToastContainer
+          position="bottom-left"
+          icon={false}
+          stacked
+          hideProgressBar
+        />
+        <div className="application-container">
+          <DataProvider runtimeConfig={runtimeConfig} session={session}>
```

Comment:
question: is there a reason we opted to swap out the default `SessionProvider` for our own custom context provider here?